### ISSUE TITLE
`UnhandledPromiseRejectionWarning` during E2E tests

### ISSUE THREAD

**Issue by rsimha** [ampproject/amphtml#27566]
This happens during E2E runs, even when all tests pass.
```
(node:8476) UnhandledPromiseRejectionWarning: AssertionError: Expected an error to be thrown.: expected undefined not to be undefined
    at _callee3$ (/home/travis/build/ampproject/amphtml/extensions/amp-carousel/0.1/test-e2e/test-hidden-controls.js:40:15)
    at tryCatch (/home/travis/build/ampproject/amphtml/build-system/tasks/e2e/node_modules/babel-regenerator-runtime/runtime.js:61:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/home/travis/build/ampproject/amphtml/build-system/tasks/e2e/node_modules/babel-regenerator-runtime/runtime.js:329:22)
    at GeneratorFunctionPrototype.prototype.<computed> [as next] (/home/travis/build/ampproject/amphtml/build-system/tasks/e2e/node_modules/babel-regenerator-runtime/runtime.js:94:21)
    at asyncGeneratorStep (/home/travis/build/ampproject/amphtml/extensions/amp-carousel/0.1/test-e2e/test-hidden-controls.js:3:103)
    at _next (/home/travis/build/ampproject/amphtml/extensions/amp-carousel/0.1/test-e2e/test-hidden-controls.js:5:194)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:8476) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 192)
(node:8476) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Example: https://travis-ci.org/github/ampproject/amphtml/jobs/670685975#L2287-L2297

**Comment by rsimha** [comment]
Tagging @ampproject/wg-ui-and-a11y since this appears to come from `extensions/amp-carousel/0.1/test-e2e/test-hidden-controls.js`

/cc @estherkim

**Comment by caroqliu** [comment]
This issue is documented in the code: https://github.com/ampproject/amphtml/blob/master/extensions/amp-carousel/0.1/test-e2e/test-hidden-controls.js#L57-L70

Happy to update the TODO with the issue number. I believe the error is a work around to convey an expectation that there aren't currently helpers to do this with.

**Comment by rsimha** [comment]
Bumping this issue since it currently makes E2E logs difficult to read, especially when there are other failures. Let us Infra folks know if we can help 😃 

![image](https://user-images.githubusercontent.com/26553114/80659892-3f2eac80-8a58-11ea-8c6c-1cddeeb56b6b.png)

**Comment by caroqliu** [comment]
These are the notes from the TODO linked prior:

> Ideally we would do something like:
> `await expect(() => controller.click(nextArrow)).to.throw();`, but
> that  does not work for a few reasons, including:
> * Click does not return a ControllerPromise
> * ControllerPromise uses a Promise rejection to signal it is not done
>    yet. The wrapper logic would need to be modified to allow for an
>    expected error type or message to sort circuit the waiting, allowing
>    use of something like:
>    ```
>    await expect(() => controller.click(nextArrow))
>       .to.throw(/ElementClickInterceptedError/);
>    ```

Let me take another look at the tests today to see if there might be alternatives such as the one you described.
